pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,384 @@
+	return map[string]string{
+		machinePoolLabelKey: machinePoolLabelValue,
+	}
+}
+
+func createMachinePoolRequestForDedicated(machinePoolNodeCount int) (*v1.MachinePool, error) {
+	mp := v1.NewMachinePool()
+	mp.ID(machinePoolId).
+		Replicas(machinePoolNodeCount).
+		InstanceType(machinePoolInstanceType).
+		Labels(createNewMachinePoolLabelsDedicated()).
+		Taints(createNewMachinePoolTaintsDedicated())
+	machinePool, err := mp.Build()
+	if err != nil {
+		return nil, err
+	}
+	return machinePool, nil
+}
+
+// TODO this function should be moved to an ocm client / provider area
+func createMachinePool(opts *options, mprequest *v1.MachinePool) error {
+	// create a new machine pool via ocm
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return err
+	}
+	client, cc, err := conn.API().OCMClustermgmt()
+	if err != nil {
+		return err
+	}
+	response, err := client.Clusters().Cluster(opts.selectedCluster.ID()).MachinePools().Add().Body(mprequest).Send()
+	if err != nil {
+		return err
+	}
+	opts.selectedClusterMachinePool = *response.Body()
+	defer cc()
+	return nil
+}
+
+func createMachinePoolInteractivePrompt(opts *options) error {
+	validator := &dedicatedcmdutil.Validator{
+		Localizer:  opts.f.Localizer,
+		Connection: opts.f.Connection,
+	}
+	// TODO add page size and Localizer
+	promptNodeCount := &survey.Input{
+		Message: "Enter the desired machine pool node count",
+		Help:    "The machine pool node count must be greater than or equal to and a multiple of 3",
+	}
+	var nodeCount int
+	err := survey.AskOne(promptNodeCount, &nodeCount, survey.WithValidator(validator.ValidatorForMachinePoolNodes))
+	if err != nil {
+		return err
+	}
+	opts.requestedMachinePoolNodeCount = nodeCount
+	dedicatedMachinePool, err := createMachinePoolRequestForDedicated(nodeCount)
+	if err != nil {
+		return err
+	}
+	err = createMachinePool(opts, dedicatedMachinePool)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// machine pool replica count must be greater than or equal and a multiple of 3
+// refactor to take in list -- select or create
+func validateMachinePoolNodes(opts *options) error {
+	for _, machinePool := range opts.existingMachinePoolList {
+
+		// manyandas code here doesn't work.
+		//var nodeCount int
+		//replicas, ok :=	machinePool.GetReplicas()
+		//if ok {
+		//	nodeCount = replicas
+		//} else {
+		//	autoscaledReplicas, ok := machinePool.GetAutoscaling()
+		//	if ok {
+		//		nodeCount = autoscaledReplicas.MaxReplicas()
+		//	}
+		//}
+
+		// does this do as expected??
+		mp, ok := machinePool.GetReplicas()
+		if !ok {
+			mp = 0
+		} else {
+			autoScaledReplicas, ok := machinePool.GetAutoscaling()
+			if ok {
+				mp = autoScaledReplicas.MaxReplicas()
+			}
+		}
+		if validateMachinePoolNodeCount(mp) &&
+			checkForValidMachinePoolLabels(machinePool) &&
+			checkForValidMachinePoolTaints(machinePool) {
+			opts.f.Logger.Info("Found a valid machine pool: %s", machinePool.ID())
```

Comment:
```suggestion
			opts.f.Logger.Infof("Found a valid machine pool: %s", machinePool.ID())
```

i guess you mean to be using `Infof`